tests/foreman/virtwho/test_api_virtwho.py

```diff
@@ -107,3 +111,243 @@ def test_positive_deploy_configure_id(self):
         vhd.delete()
         self.assertEqual(entities.VirtWhoConfig().search(
             query={'search': 'name={}'.format(name)}), [])
+
+    @tier1
+    def test_positive_deploy_configure_by_script(self):
+        """ Verify "GET /foreman_virt_who_configure/api/
+
+        v2/configs/:id/deploy_script"
+
+        :id: bb673b27-c258-4517-8fb9-436a4b51ba9d
+
+        :expectedresults: Config can be created and deployed
+
+        :CaseLevel: Integration
+
+        :CaseImportance: High
+        """
+        name = gen_string('alpha')
+        args = self._make_virtwho_configure()
+        args.update({'name': name})
+        vhd = entities.VirtWhoConfig(**args).create()
+        self.assertEqual(vhd.status, 'unknown')
+        script = vhd.deploy_script()
+        hypervisor_name, guest_name = deploy_configure_by_script(
+            script['virt_who_config_script'], debug=True)
+        status = entities.VirtWhoConfig().search(
+            query={'search': 'name={}'.format(name)}
+        )[0].status
+        self.assertEqual(status, 'ok')
+        hosts = [
+            (hypervisor_name, 'product_id={}'.format(self.vdc_physical)),
+            (guest_name, 'type = STACK_DERIVED')]
+        for hostname, sku in hosts:
+            host = entities.Host().search(
+                query={'search': "{0}".format(hostname)})[0]
+            product_subscription = entities.Subscription().search(
+                query={'search': '{0}'.format(sku)})[0]
+            entities.HostSubscription(host=host.id).add_subscriptions(
+                data={'subscriptions': [{
+                              'id': product_subscription.id,
+                              'quantity': 1}]})
+            result = entities.Host().search(
+                query={'search': '{0}'.format(hostname)})[0].read_json()
+            self.assertEqual(
+                result['subscription_status_label'],
+                'Fully entitled')
+        vhd.delete()
+        self.assertEqual(entities.VirtWhoConfig().search(
+            query={'search': 'name={}'.format(name)}), [])
+
+    @tier1
+    def test_positive_debug_option(self):
+        """ Verify debug option by "PUT
+
+        /foreman_virt_who_configure/api/v2/configs/:id"
+
+        :id: 75a20b8c-bed8-4c55-b291-14bca6cac364
+
+        :expectedresults: debug option can be updated.
+
+        :CaseLevel: Integration
+
+        :CaseImportance: Medium
+        """
+        name = gen_string('alpha')
+        args = self._make_virtwho_configure()
+        args.update({'name': name})
+        vhd = entities.VirtWhoConfig(**args).create()
+        options = {'true': '1', 'false': '0', '1': '1', '0': '0'}
```

Comment:
how about `yes` and `no`, are they available?

Reply:
According to API Recourse(API documentation) on Sat UI, the `foreman_virt_who_configure_config[debug]`  must be one of : `true`, `false`, `1`, `0`, I will try it and give feedback. Thanks for the reminder.

Reply:
no support `yes` and `no`, when I update debug with `no`, the result is ` 'debug':'true'`